api/app/src/mappings/apple/sleep.ts

```diff
@@ -0,0 +1,55 @@
+import { Sleep } from "@metriport/api";
+import dayjs from "dayjs";
+
+import { AppleHealth, AppleHealthSleepItem, SleepType, createMetadata } from ".";
+import { ISO_DATE } from "../../shared/date";
+
+export function mapDataToSleep(data: AppleHealth) {
+  const sleep: Sleep[] = [];
+
+  const addToSleep = (appleItem: AppleHealthSleepItem) => {
+    const date = dayjs(appleItem.date).format(ISO_DATE);
+
+    const sleepPayload: Sleep = {
+      metadata: createMetadata(date),
+      start_time: appleItem.date,
+      end_time: appleItem.endDate,
+    }
+
+    if (appleItem.type === SleepType.inBed) {
+      sleepPayload.durations = {
+        in_bed_seconds: appleItem.value
+      }
+    }
+
+    if (appleItem.type === SleepType.awake) {
+      sleepPayload.durations = {
+        awake_seconds: appleItem.value
+      }
+    }
+
+    if (appleItem.type === SleepType.core) {
+      sleepPayload.durations = {
+        light_seconds: appleItem.value
+      }
+    }
+
+    if (appleItem.type === SleepType.deep) {
+      sleepPayload.durations = {
+        deep_seconds: appleItem.value
+      }
+    }
+
+    if (appleItem.type === SleepType.rem) {
+      sleepPayload.durations = {
+        rem_seconds: appleItem.value
+      }
+    }
+
+    sleep.push(sleepPayload);
+  };
+
+  data.HKCategoryValueSleepAnalysis?.forEach(appleSleepItem => addToSleep(appleSleepItem));
+
+  return sleep;
+}
```

Comment:
The data for sleep in the historical payload (last 30 days) will not be sent as daily. Instead im sending it with the start and end time of each type of sleep. The reasoning is because technically someone can fall asleep at after midnight but technically that might still be considered yesterdays sleep. By sending all types individually with start and end time customers can manipulate the data as they want without risk of it being on the wrong day because it was aggregated.

Reply:
Good idea, makes sense